main/src/ca/uwaterloo/flix/language/ast/shared/SymUse.scala

```diff
@@ -36,8 +37,23 @@ object SymUse {
 
   /**
     * Represents a use of an effect sym.
+    *
+    * For an occurrence of the form `Xxx.Yyy.Zzz`, `sym` is the [[Symbol]] of the accessed element `Zzz`
+    * and `qname` represents the qualified name `Xxx.Yyy.Zzz` in its entirety, where [[QName.namespace]]
+    * contains an [[Ident]] for `Xxx` and `Yyy` and [[QName.ident]] contains a [[Ident]] for `Zzz`.
+    *
+    * @param sym    The [[Symbol]] being used.
+    * @param qname  The qualified name ([[QName]]) of the use of `sym`:
     */
-  case class EffectSymUse(sym: Symbol.EffectSym, loc: SourceLocation)
+  case class EffectSymUse(sym: Symbol.EffectSym, qname: QName) {
+    /**
+      * Returns the [[SourceLocation]] of this [[EffectSymUse]]. Note that this is
+      * the same [[SourceLocation]] as that of the [[EffectSymUse]]s `qname`.
+      *
+      * @return The [[SourceLocation]] of the [[EffectSymUse]].
+      */
+    def loc: SourceLocation = qname.loc
```

Comment:
Is this a good idea?

A use essentially has two (or more) associated locations, right?

Why is loc equal to qname.loc? It might as well be sym.loc?

Reply:
I don't think it would make sense for `sym.loc` to be the SourceLocation of the SymUse, since that is the SourceLocation of the declaration site of the symbol, but you do have a point. 

`qname` contains multiple SourceLocations that might be considered the SourceLocation of the SymUse. For instance, should it be the SourceLocation of the entire qualified name or just that of the symbols occurrence in that qualified name?

So it might be better to leave no `.loc` at all, and let whoever uses `SymUse` decide what `SourceLocation` they care about. Maybe rename `qname` to `use`, to make it clear that that is the qname of the use of the symbol.

Reply:
The def of `loc` that I gave here gives the exact same `SourceLocaiton` for the SymUse as before this PR

Reply:
Do you mean that it doesn't make much sense to choose that specific SourceLocation now that we have the entire QName?

Reply:
> So it might be better to leave no .loc at all, and let whoever uses SymUse decide what SourceLocation they care about. 

Yes. Then the consumer has to pattern match and choose the specific location they want.